randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -38,6 +38,10 @@ def _create_pickups_dict(self, pickup_list: list[ExportedPickupDetails], item_in
         for pickup in pickup_list:
             quantity = pickup.conditional_resources[0].resources[0][1]
             object_name = self.game.region_list.node_from_pickup_index(pickup.index).extra["object_name"]
+            text_index = 1 if (pickup.original_pickup.resource_lock and
+                               "Locked" in pickup.original_pickup.resource_lock.temporary_item.long_name and
```

Comment:
Maybe use a variable for `pickup.original_pickup.resource_lock`. It it's really lengthy.